### ISSUE TITLE
Failed to create shared filesystem with default example

### ISSUE THREAD

**Issue by sdnetwork** [rook/rook#2033]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Filesystem not created
**Expected behavior:**
Create a new filsystem
**How to reproduce it (minimal and precise):**

Apply default manifest rook/cluster/examples/kubernetes/ceph/filesystem.yaml

found this error in rook-operator log : 

    exec: Running command: ceph osd pool create myfs-data0 0 replicated myfs-data0 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/299116874
    exec: Error EINVAL: (22) Invalid argument

**Environment**:
* Ubuntu 16.04.4 LTS
* Kernel (e.g. `uname -a`):  4.4.0-133-generic
* Cloud provider or hardware configuration: public cloud ovh
* Rook version (use `rook version` inside of a Rook Pod): 0.8
* Kubernetes version (use `kubectl version`):1.11
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):baremetal
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/Rook/master/toolbox.html)): HEALTH_OK

**Comment by sdnetwork** [comment]
entire logs from rook-operator

    2018-08-16 12:41:23.959385 I | rookcmd: starting Rook v0.8.0-84.gc019991 with arguments '/usr/local/bin/rook ceph operator'
    2018-08-16 12:41:23.959934 I | rookcmd: flag values: --help=false, --log-level=INFO, --mon-healthcheck-interval=45s, --mon-out-timeout=5m0s
    2018-08-16 12:41:23.961742 I | cephcmd: starting operator
    2018-08-16 12:41:24.042297 I | op-agent: getting flexvolume dir path from FLEXVOLUME_DIR_PATH env var
    2018-08-16 12:41:24.042502 I | op-agent: flexvolume dir path env var FLEXVOLUME_DIR_PATH is not provided. Defaulting to: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/
    2018-08-16 12:41:24.042603 I | op-agent: discovered flexvolume dir path from source default. value: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/
    2018-08-16 12:41:24.060948 I | op-agent: rook-ceph-agent daemonset already exists, updating ...
    2018-08-16 12:41:24.079648 I | op-discover: rook-discover daemonset already exists, updating ...
    2018-08-16 12:41:24.087407 I | operator: rook-provisioner ceph.rook.io/block started using ceph.rook.io flex vendor dir
    2018-08-16 12:41:24.113696 I | operator: rook-provisioner rook.io/block started using rook.io flex vendor dir
    2018-08-16 12:41:24.113897 I | op-cluster: start watching clusters in all namespaces
    2018-08-16 12:41:24.131005 I | op-cluster: skipping watching for legacy rook cluster events (legacy cluster CRD probably doesn't exist): the server could not find the requested resource (get clusters.rook.io)
    2018-08-16 12:41:24.133940 I | op-cluster: starting cluster in namespace rook-ceph
    2018-08-16 12:41:30.154554 I | op-k8sutil: verified the ownerref can be set on resources
    2018-08-16 12:41:30.158593 I | op-mon: start running mons
    2018-08-16 12:41:30.166242 I | cephmon: parsing mon endpoints: rook-ceph-mon2=10.111.48.238:6790,rook-ceph-mon0=10.102.176.255:6790,rook-ceph-mon1=10.108.35.4:6790
    2018-08-16 12:41:30.166530 I | op-mon: loaded: maxMonID=2, mons=map[rook-ceph-mon2:0xc42050fb20 rook-ceph-mon0:0xc42050fc00 rook-ceph-mon1:0xc42050fc80], mapping=&{Node:map[rook-ceph-mon0:0xc420219e00 rook-ceph-mon1:0xc420219ef0 rook-ceph-mon2:0xc420219f20] Port:map[]}
    2018-08-16 12:41:30.175464 I | op-mon: saved mon endpoints to config map map[mapping:{"node":{"rook-ceph-mon0":{"Name":"slave1","Hostname":"slave1","Address":"142.44.164.38"},"rook-ceph-mon1":{"Name":"slave2","Hostname":"slave2","Address":"142.44.164.37"},"rook-ceph-mon2":{"Name":"slave3","Hostname":"slave3","Address":"142.44.164.34"}},"port":{}} data:rook-ceph-mon0=10.102.176.255:6790,rook-ceph-mon1=10.108.35.4:6790,rook-ceph-mon2=10.111.48.238:6790 maxMonId:2]
    2018-08-16 12:41:30.176228 I | cephmon: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
    2018-08-16 12:41:30.176563 I | cephmon: copying config to /etc/ceph/ceph.conf
    2018-08-16 12:41:30.178376 I | cephmon: generated admin config in /var/lib/rook/rook-ceph
    2018-08-16 12:41:30.580670 I | op-mgr: start running mgr
    2018-08-16 12:41:30.583485 I | op-mgr: the mgr keyring was already generated
    2018-08-16 12:41:30.596337 I | op-mgr: rook-ceph-mgr-a deployment already exists
    2018-08-16 12:41:30.596675 I | exec: Running command: ceph mgr module enable prometheus --force --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/944044009
    2018-08-16 12:41:31.822815 I | op-mgr: mgr metrics service already exists
    2018-08-16 12:41:31.823311 I | exec: Running command: ceph mgr module enable dashboard --force --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/363266612
    2018-08-16 12:41:32.817722 I | op-mgr: dashboard service already exists
    2018-08-16 12:41:32.817947 I | op-osd: start running osds in namespace rook-ceph
    2018-08-16 12:41:32.818179 I | exec: Running command: ceph osd set noscrub --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/609010691
    2018-08-16 12:41:33.846184 I | exec: noscrub is set
    2018-08-16 12:41:33.846552 I | exec: Running command: ceph osd set nodeep-scrub --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/007770502
    2018-08-16 12:41:34.820047 I | exec: nodeep-scrub is set
    2018-08-16 12:41:34.837842 I | op-osd: 3 of the 3 storage nodes are valid
    2018-08-16 12:41:34.838014 I | op-osd: checking if orchestration is still in progress
    2018-08-16 12:41:34.840568 I | op-osd: start provisioning the osds on nodes, if needed
    2018-08-16 12:41:34.847288 I | op-osd: avail devices for node slave1: []
    2018-08-16 12:41:34.856053 I | op-osd: Removing previous provision job for node slave1 to start a new one
    2018-08-16 12:41:34.918209 I | op-osd: batch job rook-ceph-osd-prepare-slave1 still exists
    2018-08-16 12:41:36.922262 I | op-osd: batch job rook-ceph-osd-prepare-slave1 deleted
    2018-08-16 12:41:36.933009 I | op-osd: osd provision job started for node slave1
    2018-08-16 12:41:36.979063 I | op-osd: avail devices for node slave2: []
    2018-08-16 12:41:36.987416 I | op-osd: Removing previous provision job for node slave2 to start a new one
    2018-08-16 12:41:37.004656 I | op-osd: batch job rook-ceph-osd-prepare-slave2 still exists
    2018-08-16 12:41:39.007854 I | op-osd: batch job rook-ceph-osd-prepare-slave2 deleted
    2018-08-16 12:41:39.015825 I | op-osd: osd provision job started for node slave2
    2018-08-16 12:41:39.043695 I | op-osd: avail devices for node slave3: []
    2018-08-16 12:41:39.077415 I | op-osd: Removing previous provision job for node slave3 to start a new one
    2018-08-16 12:41:39.106994 I | op-osd: batch job rook-ceph-osd-prepare-slave3 still exists
    2018-08-16 12:41:41.109996 I | op-osd: batch job rook-ceph-osd-prepare-slave3 deleted
    2018-08-16 12:41:41.118078 I | op-osd: osd provision job started for node slave3
    2018-08-16 12:41:41.118108 I | op-osd: start osds after provisioning is completed, if needed
    2018-08-16 12:41:41.134436 I | op-osd: osd orchestration status for node slave1 is completed
    2018-08-16 12:41:41.134456 I | op-osd: starting 1 osd daemons on node slave1
    2018-08-16 12:41:41.158593 I | op-osd: deployment for osd 2 already exists. updating if needed
    2018-08-16 12:41:41.169490 I | op-k8sutil: updating deployment rook-ceph-osd-id-2
    2018-08-16 12:41:43.193105 I | op-k8sutil: finished waiting for updated deployment rook-ceph-osd-id-2
    2018-08-16 12:41:43.193149 I | op-osd: started deployment for osd 2 (dir=true, type=)
    2018-08-16 12:41:43.199715 I | op-osd: osd orchestration status for node slave2 is completed
    2018-08-16 12:41:43.199730 I | op-osd: starting 1 osd daemons on node slave2
    2018-08-16 12:41:43.210562 I | op-osd: deployment for osd 0 already exists. updating if needed
    2018-08-16 12:41:43.214466 I | op-k8sutil: updating deployment rook-ceph-osd-id-0
    2018-08-16 12:41:45.247274 I | op-k8sutil: finished waiting for updated deployment rook-ceph-osd-id-0
    2018-08-16 12:41:45.247297 I | op-osd: started deployment for osd 0 (dir=true, type=)
    2018-08-16 12:41:45.252453 I | op-osd: osd orchestration status for node slave3 is starting
    2018-08-16 12:41:45.252471 I | op-osd: 2/3 node(s) completed osd provisioning
    2018-08-16 12:41:45.254309 I | op-osd: osd orchestration status for node slave3 is computingDiff
    2018-08-16 12:41:45.254329 I | op-osd: osd orchestration status for node slave3 is orchestrating
    2018-08-16 12:41:45.254600 I | op-osd: osd orchestration status for node slave3 is completed
    2018-08-16 12:41:45.254609 I | op-osd: starting 1 osd daemons on node slave3
    2018-08-16 12:41:45.259180 I | op-osd: deployment for osd 1 already exists. updating if needed
    2018-08-16 12:41:45.261538 I | op-k8sutil: updating deployment rook-ceph-osd-id-1
    2018-08-16 12:41:47.283723 I | op-k8sutil: finished waiting for updated deployment rook-ceph-osd-id-1
    2018-08-16 12:41:47.283765 I | op-osd: started deployment for osd 1 (dir=true, type=)
    2018-08-16 12:41:47.291259 I | op-osd: 3/3 node(s) completed osd provisioning
    2018-08-16 12:41:47.291326 I | op-osd: checking if any nodes were removed
    2018-08-16 12:41:47.298094 I | op-osd: processing 0 removed nodes
    2018-08-16 12:41:47.298110 I | op-osd: done processing removed nodes
    2018-08-16 12:41:47.298119 I | op-osd: completed running osds in namespace rook-ceph
    2018-08-16 12:41:47.298264 I | exec: Running command: ceph osd unset noscrub --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/601750829
    2018-08-16 12:41:47.741762 I | exec: noscrub is unset
    2018-08-16 12:41:47.742101 I | exec: Running command: ceph osd unset nodeep-scrub --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/989445288
    2018-08-16 12:41:48.736865 I | exec: nodeep-scrub is unset
    2018-08-16 12:41:48.737320 I | op-cluster: Done creating rook instance in namespace rook-ceph
    2018-08-16 12:41:48.746117 I | op-pool: start watching pool resources in namespace rook-ceph
    2018-08-16 12:41:48.747001 I | op-pool: skipping watching for legacy rook pool events (legacy pool CRD probably doesn't exist): the server could not find the requested resource (get pools.rook.io)
    2018-08-16 12:41:48.747141 I | op-object: start watching object store resources in namespace rook-ceph
    2018-08-16 12:41:48.747941 I | op-object: skipping watching for legacy rook objectstore events (legacy objectstore CRD probably doesn't exist): the server could not find the requested resource (get objectstores.rook.io)
    2018-08-16 12:41:48.748039 I | op-file: start watching filesystem resource in namespace rook-ceph
    2018-08-16 12:41:48.765257 I | op-file: skipping watching for legacy rook filesystem events (legacy filesystem CRD probably doesn't exist): the server could not find the requested resource (get filesystems.rook.io)
    2018-08-16 12:41:48.768525 I | op-cluster: finalizer already set on cluster rook-ceph
    2018-08-16 12:41:48.769162 I | op-cluster: update event for cluster rook-ceph
    2018-08-16 12:41:48.769246 I | op-cluster: update event for cluster rook-ceph is not supported
    2018-08-16 12:41:48.769328 I | op-cluster: update event for cluster rook-ceph
    2018-08-16 12:41:48.769424 I | op-cluster: update event for cluster rook-ceph is not supported
    2018-08-16 12:41:49.262742 I | exec: Running command: ceph osd crush dump --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/682653671
    2018-08-16 12:41:49.605911 I | exec: Running command: ceph fs get myfs --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/387926554
    2018-08-16 12:41:49.967568 I | exec: Error ENOENT: filesystem 'myfs' not found
    2018-08-16 12:41:49.967938 I | exec: Running command: ceph fs ls --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/322486449
    2018-08-16 12:41:50.371916 I | cephmds: Creating file system myfs
    2018-08-16 12:41:50.372295 I | exec: Running command: ceph osd crush rule create-simple myfs-metadata default host --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/131805788
    2018-08-16 12:41:50.710890 I | exec: rule myfs-metadata already exists
    2018-08-16 12:41:50.711390 I | exec: Running command: ceph osd pool create myfs-metadata 0 replicated myfs-metadata --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/192761867
    2018-08-16 12:41:51.800830 I | exec: pool 'myfs-metadata' created
    2018-08-16 12:41:51.801205 I | exec: Running command: ceph osd pool set myfs-metadata size 1 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/637259246
    2018-08-16 12:41:52.846431 I | exec: set pool 10 size to 1
    2018-08-16 12:41:52.846761 I | exec: Running command: ceph osd pool application enable myfs-metadata cephfs --yes-i-really-mean-it --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/800826485
    2018-08-16 12:41:53.891704 I | exec: enabled application 'cephfs' on pool 'myfs-metadata'
    2018-08-16 12:41:53.891982 I | cephclient: creating replicated pool myfs-metadata succeeded, buf: 
    2018-08-16 12:41:53.892167 I | exec: Running command: ceph osd crush rule create-simple myfs-data0 default osd --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/287462224
    2018-08-16 12:41:54.258083 I | exec: rule myfs-data0 already exists
    2018-08-16 12:41:54.258434 I | exec: Running command: ceph osd pool create myfs-data0 0 replicated myfs-data0 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/859296879
    2018-08-16 12:41:54.879128 I | exec: Error EINVAL: (22) Invalid argument
    2018-08-16 12:41:54.879425 E | op-file: failed to create file system myfs. failed to create file system myfs: failed to create data pool myfs-data0. failed to create replicated pool myfs-data0. exit status 22

**Comment by travisn** [comment]
@sdnetwork was the filesystem previous created before you created the filesystem crd? See this log statement:
```
2018-08-16 12:41:54.258083 I | exec: rule myfs-data0 already exists
```
I believe the invalid argument error happens if it already exists.

**Comment by sdnetwork** [comment]
thx, i have do that 

    ceph osd crush rule rm myfs-data0

and restart operator and it works !!

**Comment by galexrt** [comment]
@travisn Seems that the operator is either a) not cleaning up the crush rule (on deletion) or b) tries to create it even though it exists.

**Comment by galexrt** [comment]
This will be fixed by #2144.
